testsuite/MDAnalysisTests/lib/test_nsgrid.py

```diff
@@ -231,3 +231,31 @@ def test_nsgrid_probe_close_to_box_boundary():
     expected_dists = np.array([2.3689647], dtype=np.float64)
     assert_equal(results.get_pairs(), expected_pairs)
     assert_allclose(results.get_pair_distances(), expected_dists, rtol=1.e-6)
+
+
+def test_zero_max_dist():
+    # see issue #2656
+    # searching with max_dist = 0.0 shouldn't cause segfault (and infinite subboxes)
+    ref = np.array([1.0, 1.0, 1.0], dtype=np.float32)
+    conf = np.array([2.0, 1.0, 1.0], dtype=np.float32)
+
+    box = np.array([10., 10., 10., 90., 90., 90.], dtype=np.float32)
+
+    res = mda.lib.distances._nsgrid_capped(ref, conf, box=box, max_cutoff=0.0)
+
+
+@pytest.fixture()
+def u_pbc_triclinic():
+    u = mda.Universe(PDB)
+    u.dimensions = [10, 10, 10, 60, 60, 60]
+    return u
+
+
+def test_around_superposed_small_res(u_pbc_triclinic):
+    ag = u_pbc_triclinic.select_atoms('around 0.0 resid 10')
+    assert len(ag) == 0
+
+
+def test_around_superposed_large_res(u_pbc_triclinic):
+    ag = u_pbc_triclinic.select_atoms('around 0.0 resid 3')
+    assert len(ag) == 0
```

Comment:
These are a little different to Lily's tests, I think the correct answer is 0 atoms, because an Around selection won't select the thing it's around-ing (https://www.mdanalysis.org/docs/documentation_pages/selections.html#geometric)

I think the small box (0.001) is causing an infinite loop [here](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/lib/nsgrid.pyx#L587) but I'm looking into it....

Reply:
I split out the tests by residue size because the smaller one uses the brute-force method and the larger one uses the nsgrid method. `bruteforce` never seg-faulted so I'm not sure that `test_around_small_res` is needed, and it would be nice to save time on tests given #2671. 

To nitpick the `test_around_superposed_large_res` name -- the larger box means there are no more atoms in exactly the same spot anymore, so the `superposed` isn't really accurate anymore. Also, it would be helpful for future readers to rename or comment the test to make clear which method of `capped_distance` is being tested here.